WcaOnRails/app/models/country.rb

```diff
@@ -2,9 +2,29 @@
 
 class Country < ApplicationRecord
   include Cachable
+  WCA_STATES_JSON_PATH = Rails.root.to_s + "/config/wca-states.json"
   self.table_name = "Countries"
 
-  MULTIPLE_COUNTRIES_IDS = %w(XA XE XS).freeze
+  MULTIPLE_COUNTRIES = [
+    { id: 'XA', name: 'Multiple Countries (Asia)', continentId: '_Asia', iso2: 'XA' },
+    { id: 'XE', name: 'Multiple Countries (Europe)', continentId: '_Europe', iso2: 'XE' },
+    { id: 'XS', name: 'Multiple Countries (South America)', continentId: '_South America', iso2: 'XS' },
+  ].freeze
+
+  MULTIPLE_COUNTRIES_IDS = MULTIPLE_COUNTRIES.map { |c| c[:id] }.freeze
+
+  WCA_STATES = JSON.parse(File.read(WCA_STATES_JSON_PATH)).freeze
+
+  ALL_STATES = [
+    WCA_STATES["states_lists"].map do |list|
+      list["states"].map do |state|
+        state_id = state["id"] || I18n.transliterate(state["name"]).tr("'", "_")
```

Comment:
Gotcha, I saw `id` for each country in the json file, but now I know that it's gonna be changed and this makes sense =)